Apps/W1/File - Azure BLOB Storage Connector/app/src/BlobStorageConnectorImpl.Codeunit.al

```diff
@@ -0,0 +1,414 @@
+        ABSOperationResponse: Codeunit "ABS Operation Response";
+        ABSOptionalParameters: Codeunit "ABS Optional Parameters";
+    begin
+        InitBlobClient(AccountId, ABSBlobClient);
+
+        if (Path <> '') and not Path.EndsWith('/') then
+            Path += '/';
+
+        ABSOptionalParameters.Prefix(Path);
+        ABSOperationResponse := ABSBlobClient.ListBlobs(ABSContainerContent, ABSOptionalParameters);
+
+        if not ABSOperationResponse.IsSuccessful() then
+            Error(ABSOperationResponse.GetError());
+
+        ABSContainerContent.SetRange("Parent Directory", Path);
+        ABSContainerContent.SetRange("Blob Type", '');
+        if not ABSContainerContent.FindSet() then
+            exit;
+
+        repeat
+            FileAccountContent.Init();
+            FileAccountContent.Name := ABSContainerContent.Name;
+            FileAccountContent.Type := FileAccountContent.Type::Directory;
+            FileAccountContent."Parent Directory" := ABSContainerContent."Parent Directory";
+            FileAccountContent.Insert();
+        until ABSContainerContent.Next() = 0;
+    end;
+
+    /// <summary>
+    /// Creates a directory on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to send out the file.</param>
+    /// <param name="Path">The directory path inside the file account.</param>
+    procedure CreateDirectory(AccountId: Guid; Path: Text)
+    begin
+
+    end;
+
+    /// <summary>
+    /// Checks if a directory exists on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to send out the file.</param>
+    /// <param name="Path">The directory path inside the file account.</param>
+    /// <returns>Returns true if the directory exists</returns>
+    procedure DirectoryExists(AccountId: Guid; Path: Text): Boolean
+    var
+        ABSBlobClient: Codeunit "ABS Blob Client";
+        ABSContainerContent: Record "ABS Container Content";
+        ABSOperationResponse: Codeunit "ABS Operation Response";
+        ABSOptionalParameters: Codeunit "ABS Optional Parameters";
+    begin
+        if Path = '' then
+            exit(true);
+
+        InitBlobClient(AccountId, ABSBlobClient);
+        ABSOptionalParameters.Prefix(Path);
+        ABSOptionalParameters.MaxResults(1);
+        ABSOperationResponse := ABSBlobClient.ListBlobs(ABSContainerContent, ABSOptionalParameters);
+        if not ABSOperationResponse.IsSuccessful() then
+            Error(ABSOperationResponse.GetError());
+
+        exit(not ABSContainerContent.IsEmpty());
+    end;
+
+    /// <summary>
+    /// Deletes a directory exists on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to send out the file.</param>
+    /// <param name="Path">The directory path inside the file account.</param>
+    procedure DeleteDirectory(AccountId: Guid; Path: Text)
+    begin
+
```

Comment:
Missing implementation. of DeleteDirectory

Reply:
I'm currently thinking about that. Azure Blob Services are Hirachy free. The API and the Azure Storage Explorer are simulating folders based on the Blob File Name. You can not create a folder only a file taht has the Folder in its Name. The https://orchardcore.net/ framework sobled this issue by creating a dummy file that is ignored by its file api.

But I am not sure if i like this "hack".

Reply:
You're right I forgot about this part for the Azure Blob Service.
Maybe there should be at least a small comment about this behaviour.

I thought if it might be helpful to add some kind of `GetSupportedMethods` to the Interface to check if one specific feature is possible for this implementation, but this would make i much more complicated....
If it isn't clear what I mean, I was thinking about the pattern that was used for the Barcode Provider. (GetSupportedBarcodeSymbologies in https://github.com/microsoft/ALAppExtensions/blob/main/Modules/System/Barcode/src/Barcode%20Provider/Font/BarcodeFontProvider.Interface.al)

Maybe someone else has an opinion on this specfic topic, too.